core/ai.go

```diff
@@ -64,24 +86,199 @@ func ParseAIModelConfigs(config string) ([]AIModelConfig, error) {
+			if err := json.Unmarshal([]byte(config), &configs); err != nil {
+				return nil, err
+			}
+		}
 	}
 
-	models := strings.Split(config, ",")
-	for _, m := range models {
-		parts := strings.Split(m, ":")
-		if len(parts) < 3 {
-			return nil, errors.New("invalid AI model config expected <pipeline>:<model_id>:<warm>")
+	return configs, nil
+}
+
+func (n *LivepeerNode) AddAIConfigs(ctx context.Context, configs []AIModelConfig) ([]Capability, map[Capability]*Constraints, error) {
+	var aiCaps []Capability
+	constraints := make(map[Capability]*Constraints)
+
+	for _, config := range configs {
+		modelConstraint := &ModelConstraint{Warm: config.Warm, Capacity: 1}
+		pipelineCap := PipelineToCapability(config.Pipeline)
+
+		if pipelineCap > Capability_Unused {
+			// If the config contains a URL we call Warm() anyway because AIWorker will just register
+			// the endpoint for an external container
+			if config.Warm || config.URL != "" {
+				endpoint := worker.RunnerEndpoint{URL: config.URL, Token: config.Token}
+				if err := n.AIWorker.Warm(ctx, config.Pipeline, config.ModelID, endpoint, config.OptimizationFlags); err != nil {
+					return nil, nil, fmt.Errorf("Error AI worker warming %v container: %v", config.Pipeline, err)
+				}
+			}
+
+			// Show warning if people set OptimizationFlags but not Warm.
+			if len(config.OptimizationFlags) > 0 && !config.Warm {
+				glog.Warningf("Model %v has 'optimization_flags' set without 'warm'. Optimization flags are currently only used for warm containers.", config.ModelID)
+			}
+			_, ok := constraints[pipelineCap]
+			if !ok {
+				aiCaps = append(aiCaps, pipelineCap)
+				constraints[pipelineCap] = &Constraints{
+					Models: make(map[string]*ModelConstraint),
+				}
+			}
+
+			_, model_exists := constraints[pipelineCap].Models[config.ModelID]
+			if model_exists {
+				if constraints[pipelineCap].Models[config.ModelID].Warm == modelConstraint.Warm {
+					constraints[pipelineCap].Models[config.ModelID].Capacity += 1
+				} else {
+					constraints[pipelineCap].Models[config.ModelID] = modelConstraint
+				}
+			} else {
+				constraints[pipelineCap].Models[config.ModelID] = modelConstraint
+			}
+
+			n.SetBasePriceForCap("default", pipelineCap, config.ModelID, big.NewRat(config.PricePerUnit, config.PixelsPerUnit))
+
 		}
 
-		pipeline := parts[0]
-		modelID := parts[1]
-		warm, err := strconv.ParseBool(parts[3])
-		if err != nil {
-			return nil, err
+		if len(aiCaps) > 0 {
+			capability := aiCaps[len(aiCaps)-1]
+			price := n.GetBasePriceForCap("default", capability, config.ModelID)
+			glog.V(common.DEBUG).Infof("Capability %s (ID: %v) advertised with model constraint %s at price %d per %d unit (%+v)", config.Pipeline, capability, config.ModelID, price.Num(), price.Denom(), modelConstraint)
 		}
+	}
+
+	return aiCaps, constraints, nil
+}
+func (n *LivepeerNode) RemoveAIConfigs(ctx context.Context, configs []AIModelConfig) ([]Capability, map[Capability]*Constraints, error) {
```

Comment:
Is this used for anything?

Reply:
It is not, was used in my alternate implementation that allowed workers to register to the orchestrator.  This is covered by `RemoveAICapabilities` with the ai worker.